### PR TITLE
Expose Ssl::Connection to http filter and expose SAN.uri in cert. 

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

**Comment by wattli** [comment]
@mattklein123 , could you take a look of this PR? Thanks.

**Comment by mattklein123** [comment]
At a high level this looks fine, but you need tests.

**Comment by wattli** [comment]
I would like to add tests. But seems all 6 files this PR touches do not have existing _test.cc file.

**Comment by wattli** [comment]
@mattklein123 , I would like to add tests. But seems all 6 files this PR touches do not have existing _test.cc file.

**Comment by mattklein123** [comment]
The lack of _test files does not mean you don't get to add tests. You need to test your code. This can either be done in unit tests or integration tests. For the SSL connection stuff, I would probably make a new _test file modeled after:
https://github.com/lyft/envoy/blob/master/test/common/network/connection_impl_test.cc

For testing the SSL filter function addition, you can add a test to here:
https://github.com/lyft/envoy/blob/master/test/common/http/conn_manager_impl_test.cc

**Comment by wattli** [comment]
@mattklein123 It's not that I don't want to add tests. Just a little weird when I don't see any tests for the files involved in this PR. Now I realized you guys have different testing path with us, typically we put _test.cc in the same dir with the class to test.

To be honest, this is my first PR and it is not that trivial. So I decided to only expose ssl() interface and will add the actual logic/tests in follow-up PRs. How does that sound?

**Comment by mattklein123** [comment]
Even for this reduced change, you still must have a test. You should test in the connection manager filter test that you correctly return the SSL info for a mock SSL connection when you call the function from a filter.

Please also revert the file mode changes in the diff.

**Comment by wattli** [comment]
@mattklein123 please take a look. Thanks.

File: test/common/http/conn_manager_impl_test.cc [hunk 1]
```diff
@@ -833,6 +833,7 @@ TEST_F(HttpConnectionManagerImplTest, MultipleFilters) {
       .WillOnce(InvokeWithoutArgs([&]() -> Http::FilterHeadersStatus {
         EXPECT_EQ(route_config_provider_.route_config_->route_,
                   decoder_filter1->callbacks_->route());
+        EXPECT_EQ(ssl_connection_.get(), decoder_filter1->callbacks_->ssl());
```
Comment (mattklein123) [inline test/common/http/conn_manager_impl_test.cc]: The way this test is setup above, ssl_connection_.get() is going to be nullptr. Can we make this an "SSL" connection and then this test would be more useful?

Comment (wattli) [inline test/common/http/conn_manager_impl_test.cc]: I added another check in the test above with ssl connection. So we are covering both ssl being nullptr and not null.

**Pull request merged**